docs/src/usage/custom_metal_kernels.rst

```diff
@@ -0,0 +1,124 @@
+.. _custom_metal_kernels:
```

Comment:
I suggest we put this in the [developer guides](https://github.com/ml-explore/mlx/tree/main/docs/src/dev) as it's not really intended for the typical user. Also I think that makes it easier to change the API if we need the flexibility while we are understanding the usage patterns.

Reply:
Definitely agree. I've moved it.